keps/sig-node/4939-tls-in-grpc-probe/README.md

```diff
@@ -0,0 +1,883 @@
+
+A limitation in the current implementation is that it only supports
+gRPC servers that do not leverage TLS connections.
+Even if they are not concerned about certificate verification for the health check,
+a connection cannot be established at all if the server is expecting TLS and the client
+is not.
+
+This enhancement aims to add configuration options to enable TLS on the gRPC probe.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+We often deploy internal gRPC services on our cluster.
+These deployments provide internal services and it's simple to add
+the health server to them so they can be verified through a single interface.
+
+It's also worth noting we have and internal CA that signs certs for communicating
+with these servers and all of them use TLS.
+
+Currently, we are using the `exec` probe for readiness and liveness configured as:
+
+```
+liveness_probe {
+  exec {
+    command = [
+      "/bin/grpc_health_probe",
+      "-addr=:8443",
+      "-tls",
+      "-tls-no-verify",
+    ]
+  }
+}
+```
+
+We would really like to switch to the gRPC probes introduced in 1.24
+but are unable to do so since there is no way to configure it to use a TLS connection
+when reaching out to the health server.
+
+Instead we must continue to rely on the `exec` probe and cannot reap the benefits
+described in [the announcement](https://kubernetes.io/blog/2022/05/13/grpc-probes-now-in-beta/).
+
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+The primary goal is to support TLS connections when using the `grpc` probe.
+The probe will use TLS but not verify the certificate.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+It is not a goal of this KEP to support providing a certificate to verify the TLS
+connection.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+I would like to add new configuration fields alongside `port` and `service` in the 
+[Probe GRPCAction](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#Probe).
+
+They can be used to indicate whether or not TLS should be used.
+
+### User Stories (Optional)
```

Comment:
Either fill these optional section in or remove them :)

Reply:
removed for now